src/shogun/lib/SGMatrix.h

```diff
@@ -306,9 +308,66 @@ template<class T> class SGMatrix : public SGReferencedData
 		/** Set matrix to a constant */
 		void set_const(T const_elem);
 
-		/** fill matrix with zeros */
+		/** Fill matrix with zeros */
 		void zero();
 
+		/**
+		 * Fill matrix with uniformly distributed randoms
+		 * Float Types: [0, 1)
+		 * Int Types: std::numeric_limit<T>::(min->max)
+		**/
+		// vectorized available
+		template <typename U>
+		using enable_simd_float = std::enable_if_t<is_any_of_v<U, float64_t>>;
```

Comment:
Just out of curiosity, `std::is_same_v` would make it here as well instead of `is_any_of_v`, yes? No need to update anything.

Reply:
Yep it would, I just kept it for consistency. Also `is_same_v` isnt part of STL till C++17. So I would have to add that too.

Reply:
I added a `sg_is_same_v` for now and have used that. Guess it might be useful at other places too and can be replaced by the STL equivalent when you'll decide to bump up the C++ version.

Reply:
yes, thx for that, maybe (if not yet done) put a comment that indicated that this can be removed/replaced